src/map.cpp

```diff
@@ -6910,6 +6910,112 @@ void map::restock_fruits( const tripoint &p, int time_since_last_actualize )
     }
 }
 
+void map::produce_sap( const tripoint &p, int time_since_last_actualize )
+{
+    if( time_since_last_actualize <= 0 ) {
+        return;
+    }
+
+    if( ter_at( p ).id != "t_tree_maple_tapped" ) {
```

Comment:
Also, not a problem here, just noting: In the future, this could be easily extended to other saps by having the sap be a terrain property.